common/src/main/java/org/esbtools/message/admin/common/EsbMessageAdminServiceImpl.java

```diff
@@ -18,94 +18,83 @@
+import static org.esbtools.message.admin.common.config.EMAConfiguration.getPartiallyViewableMessages;
+import static org.esbtools.message.admin.common.config.EMAConfiguration.getResyncRestEndpoints;
+import static org.esbtools.message.admin.common.config.EMAConfiguration.getSortingFields;
+import static org.esbtools.message.admin.common.config.EMAConfiguration.getSuggestedFields;
+
 @Named
 public class EsbMessageAdminServiceImpl implements Provider {
 
     private static final Logger LOG = LoggerFactory.getLogger(EsbMessageAdminServiceImpl.class);
-    private JSONObject config;
-    private String encryptionKey;
-    private transient EsbErrorDAO errorDao;
-    private transient MetadataDAO metadataDao;
-    private transient AuditEventDAO auditDao;
+    private static final String ERROR_KEY_TYPE = "error";
+    private static final String MESSAGE_PROPERTY_PAYLOAD_HASH = "esbPayloadHash";
+    private static final String ILLEGAL_ARGUMENT = "Illegal Argument:";
+    private static final String DEFAULT_USER = "someUser";
+    private static final String METADATA_KEY_TYPE = "metadata";
+    private static final String TYPE_PLACEHOLDER = "$TYPE";
+    private static final String METADATA_QUERY = "select f from MetadataEntity f where f.type = '" + TYPE_PLACEHOLDER + "'";
     private static transient KeyExtractorUtil extractor;
-    private static transient EncryptionUtil encrypter;
-
-    private static final String DEFAULT_ENCODING = "UTF-8";
+    private static transient EncryptionUtility encryptor;
+    private static transient Map<MetadataType, MetadataResponse> treeCache = new ConcurrentHashMap<>();
+    private static transient Map<String, List<String>> suggestionsCache = new ConcurrentHashMap<>();
 
     @Inject
     private EntityManager entityMgr;
 
-    public EsbMessageAdminServiceImpl() {
-        try {
-            InputStream configFile = this.getClass().getClassLoader().getResourceAsStream("config.json");
-            JSONParser parser = new JSONParser();
-            config = (JSONObject) parser.parse(new InputStreamReader(configFile, DEFAULT_ENCODING));
-            configFile.close();
-            InputStream encryptionKeyFile = this.getClass().getClassLoader().getResourceAsStream("encryption.key");
-            BufferedReader encryptionKeyFileReader = new BufferedReader(new InputStreamReader(encryptionKeyFile, DEFAULT_ENCODING));
-            encryptionKey = encryptionKeyFileReader.readLine();
-            encryptionKeyFileReader.close();
-            encryptionKeyFile.close();
-        } catch (Exception e) {
-            throw new IllegalStateException("Unable to load configuration files", e);
-        }
-    }
-
-    private EsbErrorDAO getErrorDAO() {
-        return errorDao == null ? new EsbErrorDAOImpl(entityMgr, getAuditEventDAO(), getEncrypter(), config) : errorDao;
-    }
-
     void setErrorEntityManager(EntityManager entityMgr) {
```

Comment:
IMHO I would move the entityMgr to the constructor and `@Inject` the constructor if CDI let's you do this (I think it does)